lib/fog/openstack/models/network/floating_ip.rb

```diff
@@ -31,9 +31,34 @@ def update
 
         def destroy
           requires :id
-          service.delete_floating_ip(self.id)
+          service.delete_floating_ip(id)
           true
         end
+
+        def associate(port_id, fixed_ip_address = nil)
+          requires :id
+          options = if !fixed_ip_address.nil?
```

Comment:
I'm not usually a fan of inline conditionals, but I think it'd be more readable here:

```ruby
options = fixed_ip_address ? {'fixed_ip_address' => fixed_ip_address} : {}
```

Since it's also used in `disassociate`, I'd put it in a private method like so

```ruby
private

def options(fixed_ip_address)
  fixed_ip_address ? {'fixed_ip_address' => fixed_ip_address} : {}
end
```

then

```ruby
merge_attributes(service.associate_floating_ip(id, port_id, options(fixed_ip_address)).body['floatingip'])
```